test/ext/sse.js

```diff
@@ -0,0 +1,186 @@
+describe("sse extension", function() {
+
+    function mockEventSource() {
+        var listeners = {};
+        var wasClosed = false;
+        var url;
+        var mockEventSource = {
+            removeEventListener: function(name) {
+                delete listeners[name];
+            },
+            addEventListener: function(message, l) {
+                listeners[message] = l;
```

Comment:
`addEventListener` should allow to register multiple event listeners for a single event, so `listeners` should store arrays, not just listener functions. I don't believe it has lead to any bugs yet, but better be safe

Reply:
Should I change this in the legacy sse tests I filched this from?

Reply:
If not tests break, it would be great. If tests break, please report it and we shall fix them or report them separately

Reply:
Should be resolved by my latest commit